pontoon/base/models/entity.py

```diff
@@ -861,14 +862,16 @@ def for_project_locale(
                 Q(
                     **{
                         f"translation__string__{translation_case_lookup}": (
-                            search,
+                            s,
                             locale.db_collation,
                         )
+                        if not search_match_case
+                        else s
                     }
                 )
```

Comment:
At this point it's more readable if you drop `translation_case_lookup` and have just one `if - else` block with the same condition:

```python
Q(translation__string__contains=s) if search_match_case else
Q(translation__string__icontains_collate=(s, locale.db_collation))
```